wgpu-core/src/device/queue.rs

```diff
@@ -371,7 +371,9 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
         let (selector, dst_base, texture_format) =
             extract_texture_selector(destination, size, &*texture_guard)?;
         let format_desc = texture_format.describe();
-        let (_, bytes_per_array_layer) = validate_linear_texture_data(
+        //Note: `_source_bytes_per_array_layer` is ignored since we have a staging copy,
+        // and it can have a different value.
+        let (_, _source_bytes_per_array_layer) = validate_linear_texture_data(
```

Comment:
Why aren't we just wholesale ignoring this value (we can leave the comment).

Reply:
We could do this. I felt that referring to a field in the comment makes it more clear if the field exists in the code.